spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingProperties.java

```diff
@@ -54,7 +54,7 @@
 
 	// Properties for both inbound/outbound
 
-	private String contentType;
+	private String contentType = "application/json";
```

Comment:
We can probably use constant from ``MimeTypeUtils`

Reply:
In case of `MessageChannel` bound components, this means that `MessageConverterConfigurer`'s  `ContentTypeConvertingInterceptor` always gets added to the corresponding `MessageChannel`. I guess this is by design of having `contentType` available all the time at the Messsage and we go by `application/json` as the default.

In case of non- message channel bound components (KStream etc,), having `contentType=application/json` as the binding property may be confusing?

Reply:
Why? Shouldn't we provide consistency across all binders? I need to sync with Soby on the KStream binder, but I believe that it could use the same logic for message converters as it's not SI specific.

I agree this is not the best place to put the default, the proper way should probably be a an EnvironmentPostProcessor, but having the same default for all binders should something we should strive for, having consistency for our users.

Reply:
Why is it bad? Spring Boot has default values in many places of the `@ConfigurationProperties`, for example: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java#L256